src/timedata.cpp

```diff
@@ -40,16 +40,20 @@ static int64_t abs64(int64_t n)
     return (n >= 0 ? n : -n);
 }
 
+#define BITCOIN_TIMEDATA_MAX_SAMPLES 200
+
 void AddTimeData(const CNetAddr& ip, int64_t nOffsetSample)
 {
     LOCK(cs_nTimeOffset);
     // Ignore duplicates
     static set<CNetAddr> setKnown;
+    if (setKnown.size() == BITCOIN_TIMEDATA_MAX_SAMPLES)
+        return;
```

Comment:
Please read the comment just after the change in the original code: https://github.com/bitcoin/bitcoin/blob/master/src/timedata.cpp#L56

This PR does not change the behaviour described there at all. I do not even plan to change the current "use the timedata samples from the first 200 peers we have seen" to "use the timedata samples from the last 200 peers we have seen" because of the comments there...

The purpose of this PR is to stop saving both `CNetAddr` for 200+th peers and rotating timedata samples because they will not be used at all anyway.